### ISSUE TITLE
Compatibility with Batch Normalization

### ISSUE THREAD

**Issue by andreped** [andreped/GradientAccumulator#6]
Current implementation of GA is not compatible with GA as BN is updated for every single micro-batch and not when the full batch is finished.

Need a way to accumulate mean/std paramemters in BN, similarly as done with gradients.

Perhaps it is possible to wrap the BN layer to add this accumulation?

**Comment by andreped** [comment]
Just came across this, which can be used for get BN compatible with GA in distributed strategy (multi-GPU training):
https://www.tensorflow.org/api_docs/python/tf/keras/layers/experimental/SyncBatchNormalization

However, it is likely possible to use this for single-GPU in GA setting as well.

**Comment by andreped** [comment]
Since batch normalization is not that suitable for larger batch sizes, we have gone with a different solution for now with `adaptive gradient clipping`. We added support for AGC in fd658989dfc481f047416f772f73cdcf60a68687.

Hence, instead of using any normalization techniques in your neural network, you should use AGC. To enable it, simply do:
```
model = GAModelWrapper(accum_steps=4, use_acg=True, clip_factor=0.01, eps=1e-3, inputs=model.input, outputs=model.output)
```
